src/Framework/Sdk/SdkResultItem.cs

```diff
@@ -55,14 +60,14 @@ public override bool Equals(object obj)
         public override int GetHashCode()
         {
             int hashCode = -849885975;
-            hashCode = (hashCode * -1521134295) + EqualityComparer<string>.Default.GetHashCode(ItemSpec);
+            hashCode = hashCode ^ ItemSpec.GetHashCode();
```

Comment:
CMIIW, I think my latest update now ensure that the key-value in the metadata ignore the orders of the key but still has the same hashcode returned.

Am I still missing something else? I welcome suggestions 🙂

Reply:
If GetHashCode() considered only the ItemSpec and ignored the Metadata, would that improve performance (because of simpler computation) or hurt performance (because of more frequent hash collisions)?

Reply:
I think we should consider GetHashCode for Metadata too. 
Now finding the best way to handle this.